xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -396,6 +393,12 @@ viewers.Comments = Class.create({
 
                 // Notify any displayed CAPTCHA that it was reloaded and it might need to reinitialize its JS.
                 this.container.fire('xwiki:captcha:reloaded');
+
+                // We send success notification only when everything is done: our integration tests relies on it
+                // for waiting a comment added.
```

Comment:
Note that I don't think this is just for our tests but for anyone listening to this event.

Reply:
Well here it's a UI event, not even a JS event.

Reply:
Indeed. However it's probably better to display that event to the user only when the action is fully done, so I believe what you did is improving that.